src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.Tests/ServiceProviderContainerTests.cs

```diff
@@ -1088,6 +1088,21 @@ public async Task GetRequiredService_ResolveUniqueServicesConcurrently_StressTes
             }
         }
 
+        [Fact]
+        public void ScopedServiceResolvedFromSingletonAfterCompilation()
```

Comment:
Do we also want to test a use case where a scoped service is registered in a scoped service provider and is depending on a singleton service?

Reply:
With the same loop? I want to say I'm sure we have that scenario covered, but never say never 😄